File: hw/pastor/u3/hw28PrinciplesAplication/PYTHON/service/service_impl.py
from service.i_service import IService
from model.generic_entity import GenericEntity

class CalculateService(IService):
    def __init__(self, repository):
        self.repository = repository

    def save_new_item(self, input_data: dict):
        entity = GenericEntity("product")

        for key, value in input_data.items():
            entity.set_data(key, value)
        
        self.repository.create(entity)

    def get_processed_data(self):
        items = self.repository.read_all()
        
        for item in items:
            try:
                price_base = float(item.get_data("priceBase"))
                tax = price_base * 0.15
                final_price = price_base + tax
                
                item.set_data("endPrice", final_price)
            except (TypeError, ValueError):
                pass 
        
        return items

File: hw/pastor/u3/hw28PrinciplesAplication/JAVASCRIPT/service/serviceImpl.js
const IGenericService = require('./iService');
const GenericEntity = require('../model/genericEntity');

class CalculateService extends IGenericService {
    constructor(repository) {
        super();
        this.repository = repository;
    }

    async saveNewItem(dataMap) {
        const entity = new GenericEntity("product");
        
        Object.keys(dataMap).forEach(key => {
            entity.setData(key, dataMap[key]);
        });

        await this.repository.create(entity);
    }

    async getProcessedData() {
        const items = await this.repository.readAll();

        items.forEach(item => {
            const priceBase = parseFloat(item.getData("priceBase"));
            
            if (!isNaN(priceBase)) {
                const finalPrice = priceBase * 1.15;
                item.setData("endPrice", finalPrice.toFixed(2));
            }
        });

        return items;
    }
}

module.exports = CalculateService;